src/bvals/bvals_interfaces.hpp

```diff
@@ -284,6 +284,53 @@ class BoundaryVariable : public BoundaryCommunication, public BoundaryBuffer {
   // private:
 };
 
+//----------------------------------------------------------------------------------------
+// Concrete classes
+//----------------------------------------------------------------------------------------
+
+//----------------------------------------------------------------------------------------
+//! \class BoundarySwarm
+class BoundarySwarm : public BoundaryCommunication {
+ public:
+  explicit BoundarySwarm(std::weak_ptr<MeshBlock> pmb);
+  ~BoundarySwarm() = default;
+
+  std::vector<ParArrayND<int>> vars_int;
+  std::vector<ParArrayND<Real>> vars_real;
+
+  // (usuallly the std::size_t unsigned integer type)
+  std::vector<BoundaryVariable *>::size_type bswarm_index;
+
+  // BoundaryCommunication
+  void SetupPersistentMPI() final;
+  void StartReceiving(BoundaryCommSubset phase) final{};
+  void ClearBoundary(BoundaryCommSubset phase) final{};
+  void Receive(BoundaryCommSubset phase);
+  void Send(BoundaryCommSubset phase);
+
+  BoundaryData<> bd_var_;
+  std::weak_ptr<MeshBlock> pmy_block;
+  Mesh *pmy_mesh_;
+  int send_tag[56], recv_tag[56];
+  int particle_size, send_size[56], recv_size[56];
```

Comment:
56 is a magic number, can you define this somewhere.

Reply:
And why did you pick 56?

Reply:
I think 56 is the maximum number of neighbors you can have if all your edge, corner, and face neighbors are refined by one AMR level. Once again I point my finger at the previous code e.g. 
```c++
template <int n = 56>                                                                                
struct BoundaryData`
```
Now might be a good time to change this though. What about `#define NMAX_NEIGHBORS 56` in `defs.hpp`? I would say `constexpr int nmax_neighbors = 56;` if we were on C++17

Reply:
I think that is a good idea.

Reply:
Done :+1: